DEVELOP.md

```diff
@@ -0,0 +1,138 @@
+# How to work on CoreCrypto
+
+## Environment
+
+<https://github.com/wireapp/core-crypto#building>
+
+### Git workflow
+
+* No merge commits. Always rebase on top of `develop`.
+* Use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) - Those are picked up by the changelog generator
+* For JIRA tickets you can add them to either the titles of PRs or the commits themselves with the following format: `[TICKET_ID]`
+
+### Versioning
+
+The versioning scheme used is [SemVer AKA Semantic Versioning][semver].
+
+## What is it?
+
+Basically, CoreCrypto is a (big) wrapper around `OpenMLS` that tries to erase the API and uphold proper invariants.
+We also take care of storage of key material on all platforms (see [docs/KEYSTORE_IMPLEMENTATION.md](docs/KEYSTORE_IMPLEMENTATION.md)).
+Another piece that is quite important is our implementation of E2EI (End-to-end-identity), see internal documents for this.
+
+## Contributing
+
+### Adding new APIs
+
+1. Make your changes wherever applicable
+2. Make sure your new API is available on `MlsCentral`, while respecting encapsulation
+    - For example, adding `MlsConversation::hello()` would mean exposing a new `MlsCentral::conversation_hello(conversation_id: ConversationId)`
+3. Expose your new API on both `crypto-ffi/src/[generic|wasm].rs`
+4. Add the new APIs respecting the appropriate calling conventions defined in [docs/FFI.md](docs/FFI.md) to
+    - Kotlin/Android: `crypto-ffi/bindings/jvm/src/main/com/wire/crypto/client/[CoreCryptoCentral|E2eiClient|MLSClient].kt`
+    - Swift/iOS: `crypto-ffi/bindings/swift/Sources/CoreCrypto/CoreCrypto.swift`
```

Comment:
This is no longer accurate since we're not tracking that file so we should just drop this line.